### PR TITLE
Refactor Scala version check and add some tests for unsupported Scala versions

### PR DESCRIPTION
**Pull request by jozic**

### CODE REVIEW

**Comment by jozic** [comment]
@ckipp01 if you have couple minutes, please check it out

File: src/main/java/org/scoverage/plugin/ScalaVersion.java
Comment (ckipp01) [inline src/main/java/org/scoverage/plugin/ScalaVersion.java:1]: This is all much nicer to work with. I know you have it covered with a bunch of itest, nice job adding those, but I also always find it worth the time to unit test classes like these as you often get way more coverage that way, and it's easy to miss something. But it's not a blocker.

**Review by ckipp01 — APPROVED** [review]
Nice clean up!

File: src/main/java/org/scoverage/plugin/ScalaVersion.java
Comment (jozic) [reply src/main/java/org/scoverage/plugin/ScalaVersion.java:1]: Thanks for the review!
I was on the fence myself with adding unit tests or not as there were none here , but after your comment I decided that it won't hurt to have those as well :)

**Pull request merged**